app/Http/Controllers/Device/Tabs/NeighboursController.php

```diff
@@ -52,6 +55,86 @@ public function name(): string
 
     public function data(Device $device): array
     {
-        return [];
+        $selection = Url::parseOptions('selection', 'list');
+
+        $links = [];
+        $devices = [];
+
+        if ($selection == 'list') {
+            $linkQuery = Port::select(
```

Comment:
You're fighting against the framework. 
Add a relationship for links here https://github.com/librenms/librenms/blob/master/app/Models/Port.php#L359
And query it like device instead

Reply:
I assumed that there was a reason that the links hadn't been added to the framework.  If this is incorrect, I will have a look at adding the links to the model.  This will also resolve both the items you've raised below which are workarounds for not having the relationships defined in the framework.

Can you please confirm that there isn't any reason you're aware of that would stop the links from being added as a model?  Is there something else that I need to clean up (e.g. an old database schema definition), or anything I need to be careful of to avoid a loss of existing data?  Is this a trickier than normal relationship (e.g. where the remote device id is allowed to be null), and possibly something I will need help with to get correct?